QRCoder/QRCodeGenerator.cs

```diff
@@ -894,35 +897,70 @@ private static string ConvertToIso8859(string value, string Iso = "ISO-8859-2")
         /// </remarks>
         private static BitArray PlainTextToBinaryByte(string plainText, EciMode eciMode, bool utf8BOM, bool forceUtf8)
         {
-            byte[] codeBytes;
+            Encoding targetEncoding;
 
             // Check if the text is valid ISO-8859-1 and UTF-8 is not forced, then encode using ISO-8859-1.
             if (IsValidISO(plainText) && !forceUtf8)
-                codeBytes = Encoding.GetEncoding("ISO-8859-1").GetBytes(plainText);
+            {
+                targetEncoding = _iso8859_1;
+                utf8BOM = false;
+            }
             else
             {
                 // Determine the encoding based on the specified ECI mode.
                 switch (eciMode)
                 {
                     case EciMode.Iso8859_1:
                         // Convert text to ISO-8859-1 and encode.
-                        codeBytes = Encoding.GetEncoding("ISO-8859-1").GetBytes(ConvertToIso8859(plainText, "ISO-8859-1"));
+                        targetEncoding = _iso8859_1;
+                        utf8BOM = false;
                         break;
                     case EciMode.Iso8859_2:
+                        // Note: ISO-8859-2 is not natively supported on .NET Core
+                        //
+                        // Users must install the System.Text.Encoding.CodePages package and call Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
+                        // before using this encoding mode.
+                        if (_iso8859_2 == null)
+                            _iso8859_2 = Encoding.GetEncoding(28592); // ISO-8859-2
                         // Convert text to ISO-8859-2 and encode.
-                        codeBytes = Encoding.GetEncoding("ISO-8859-2").GetBytes(ConvertToIso8859(plainText, "ISO-8859-2"));
```

Comment:
The `Encoding.GetBytes` method would remove characters that do not fit in the target encoding anyway, so the call to `ConvertToIso8859` was redundant.